node/core/prospective-parachains/src/fragment_tree.rs

```diff
@@ -53,6 +58,55 @@
 //! Our assumption is that the amount of candidates and parachains we consider will be reasonably
 //! bounded and in practice will not exceed a few thousand at any time. This naive implementation
 //! will still perform fairly well under these conditions, despite being somewhat wasteful of memory.
+//!
+//! # Main Types
```

Comment:
Included some notes I took. I have a feeling this change will be rejected, but it helped a lot with my comprehension. I can instead convert it to prose?

Reply:
This seems fine to me. We push back on implementation details in the guide, since other client developers may choose to implement the client with different type names, but here in the codebase I think it's helpful.

Reply:
Yup, I get that. Maybe I can move some of this into the overview and the rest into docstrings. No problem, I already had these notes so it's not wasted effort to remove this section.